apps/emqx/src/emqx_maybe.erl

```diff
@@ -0,0 +1,42 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2017-2023 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+-module(emqx_maybe).
```

Comment:
Nit: Seems that only one fun from the introduced module is really used 😅 
We should cover the others with some tests if we want to keep them.

Reply:
That's true. Tests are incoming. Thought of switching all users of `emqx_misc:maybe_apply/2` to `emqx_maybe:apply/2` but hadn't gathered enough courage. 🌝

Reply:
Made 2 commits: one with tests, another with users switch, to be able to pull or drop them separately if needed.

Reply:
Maybe we should made a separate PR with swithcing the users — just to undurden the team while reviewing the whole file transfer PR that is going to be quite large?

Reply:
Backed the commit out.